module/VuFind/src/VuFind/Http/ServerUrlHelper.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * Server URL Helper class.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFind\Http;
+
+use Closure;
+use Laminas\View\Renderer\PhpRenderer;
+
+/**
+ * Server URL Helper class.  Wrapper around Laminas ServerUrlHelper.
+ *
+ * @category VuFind
+ * @package  Http
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class ServerUrlHelper
+{
+    /**
+     * Server URL helper
+     */
+    protected Closure $serverUrlHelper;
+
+    /**
+     * Constructor.
+     *
+     * @param PhpRenderer $viewRenderer View renderer
+     *
+     * @return void
+     */
+    public function __construct(PhpRenderer $viewRenderer)
+    {
+        $this->serverUrlHelper = Closure::fromCallable($viewRenderer->plugin('serverurl'));
+    }
+
+    /**
+     * Calls Laminas ServerUrl helper to return a fully qualified URL based on the current host.
+     *
+     * @param string|bool|null $requestUri [optional] if true, the request URI
+     *                                     found in $_SERVER will be appended
+     *                                     as a path. If a string is given, it
+     *                                     will be appended as a path. Default
+     *                                     is to not append any path.
```

Comment:
I wouldn't expose the implementation details in the comments, since they may change in future, and the goal here is to create a more stable interface for the future.

Additionally, the comment lines can be unwrapped.

```suggestion
     * Return a fully qualified URL based on the current host.
     *
     * @param string|bool|null $requestUri [optional] if true, the request URI
     * found in $_SERVER will be appended as a path. If a string is given, it
     * will be appended as a path. Default is to not append any path.
```

Reply:
The thing is, this interface is ugly -- string|bool|null?  And false isn't even considered?  And the descriptions of what actually happens is so unclear that I have to look to the code to make sense of it.  I kept the reference to Laminas and the ugly-formatted code, as a hint (too subtle though) that it's not our fault. 😉  If the goal is to be future-proof, maybe we should think up a better interface, at the cost of making the wrapping less trivial?

Reply:
Fair enough, I'm certainly open to putting multiple methods on the helper instead of overloading it -- like `getBaseUrl`, `getCurrentUrl` and `getUrlForPath` maybe? Then calls can look like `$serverUrlHelper->getBaseUrl()`, etc.

Reply:
Finally did this refactoring.